platform/src/main/java/org/stellar/anchor/platform/service/TransactionService.java

```diff
@@ -131,17 +131,27 @@ public JdbcSepTransaction findTransaction(String txnId) throws AnchorException {
   public PatchTransactionsResponse patchTransactions(PatchTransactionsRequest request)
       throws AnchorException {
     List<PatchTransactionRequest> patchRequests = request.getRecords();
-
     List<GetTransactionResponse> txnResponses = new LinkedList<>();
-    for (PatchTransactionRequest patchRequest : patchRequests) {
-      txnResponses.add(patchTransaction(patchRequest));
+
+    try {
```

Comment:
I think body should be validated instead of catching NPE
With body validation, proper message can be created "Missing field X in the request"

Reply:
Yeah, I talked to @lijamie98 about that and he asked me to validate in the service and not in the controller, that is why I did it that way. Also is better to test.

But I can do something like.
```
if(patchRequests == null){
  throw BadRequestException();
}
```

For me is pretty much the same. 
WDYT?

Reply:
Updated